### PR TITLE
Add esm-env and replace browser imports

### PR DESCRIPTION
**Pull request by laurakwhit**
## Description & motivation 💭 

Screenshot 2023-04-13 at 5 47 43 PM

Instead of using `import { browser } from '$app/environment'` which is tied to SvelteKit, use `import { BROWSER } from 'esm-env'` so packages can work with any Svelte project.

### CODE REVIEW

**Review by rossedfort — APPROVED** [review]
lgtm, can we add a lint rule to prevent us from importing from `$app/environment` ? Or will there be an obvious error/failure if we do?

**Comment by laurakwhit** [comment]
It will not currently error if we use `$app/environment`. We could definitely add a lint rule with a suggestion. Do we have any precedent for custom lint rules?

**Comment by rossedfort** [comment]
lgtm 💯

File: .eslintrc.cjs [hunk 1]
```diff
@@ -68,5 +68,16 @@ module.exports = {
       'error',
       { assertionStyle: 'as' },
     ],
+    '@typescript-eslint/no-restricted-imports': [
+      'error',
+      {
+        paths: [
+          {
+            name: '$app/environment',
+            message: 'Please use esm-env instead.',
```
Comment (Alex-Tideman) [inline .eslintrc.cjs:77]: 🪄

**Pull request merged**